Spigot-Server-Patches/0139-Implement-Profile-api.patch

```diff
@@ -1,9 +1,157 @@
-From 79641217a46ed94d98a48eb658002a468a51cbf4 Mon Sep 17 00:00:00 2001
+From 49516ce6d9185f81290903aabcc4c34b2ffd0e10 Mon Sep 17 00:00:00 2001
 From: Techcable <Techcable@outlook.com>
 Date: Sun, 13 Mar 2016 20:42:57 -0700
 Subject: [PATCH] Implement Profile api.
 
 
+diff --git a/src/main/java/com/destroystokyo/paper/profile/EventProfileLookup.java b/src/main/java/com/destroystokyo/paper/profile/EventProfileLookup.java
+new file mode 100644
+index 0000000..485820e
+--- /dev/null
++++ b/src/main/java/com/destroystokyo/paper/profile/EventProfileLookup.java
+@@ -0,0 +1,141 @@
++package com.destroystokyo.paper.profile;
++
++import java.util.ArrayList;
++import java.util.Collection;
++import java.util.List;
++import java.util.UUID;
++
++import com.destroystokyo.paper.profile.event.AsyncPropertiesPreResolveEvent;
++import com.google.common.base.Preconditions;
++
++import org.bukkit.Bukkit;
++import com.destroystokyo.paper.profile.event.AsyncNamePreResolveEvent;
++import com.destroystokyo.paper.profile.event.AsyncProfileResolveEvent;
++import com.destroystokyo.paper.profile.event.AsyncUUIDPreResolveEvent;
++
++public class EventProfileLookup implements ProfileLookup {
++    private final ProfileLookup delegate;
++
++    public EventProfileLookup(ProfileLookup delegate) {
++        Preconditions.checkNotNull(delegate, "Null delegate");
++        this.delegate = delegate;
```

Comment:
@kashike That isn't a normal syntax used in the project nor necessary.

Best to leave as is. It reads cleaner as the way it currently is.